packages/legacy/core/App/screens/ProofRequestDetails.tsx

```diff
@@ -224,10 +224,12 @@ const ProofRequestDetails: React.FC<ProofRequestDetailsProps> = ({ route, naviga
     { visible: boolean; predicate: string | undefined } | undefined
   >(undefined)
 
-  useEffect(() => {
-    const template = getProofRequestTemplate(templateId)
-    if (!template) return
+  const template = useTemplate(templateId)
+  if (!template) {
+    throw new Error('Unable to find proof request template')
```

Comment:
I think this should be a `BifoldError` - no?

Reply:
What is the rule for choosing the error kind?
If the template is not found, it's unexpected behavior as we select it on the previous screen. 
I looked to other screens for credentials and proof  request details and they threw similar errors
```
throw new Error('Unable to fetch proof from AFJ')
throw new Error('Unable to fetch credential from AFJ')
```